src/main/webapp/app/exercises/quiz/manage/quiz-exercise-update.component.ts

```diff
@@ -0,0 +1,594 @@
+            const endTime = startTime.add(dayjs.duration(this.duration.minutes, 'minutes')).add(dayjs.duration(this.duration.seconds, 'seconds'));
+            batch.startTimeError = startTime.isBefore(this.quizExercise.releaseDate) || (dueDate != undefined && endTime.isAfter(dueDate));
+        });
+    }
+
+    cacheValidation() {
+        if (this.quizExercise.quizMode === QuizMode.SYNCHRONIZED) {
+            this.quizExercise.dueDate = undefined; // Due date is calculated on server side
+            if (this.scheduleQuizStart) {
+                if ((this.quizExercise.quizBatches?.length ?? 0) !== 1) {
+                    this.quizExercise.quizBatches = [this.quizExercise.quizBatches?.[0] ?? new QuizBatch()];
+                }
+            } else {
+                if ((this.quizExercise.quizBatches?.length ?? 0) !== 0) {
+                    this.quizExercise.quizBatches = [];
+                }
+            }
+        }
+
+        this.validateDate();
+        return super.cacheValidation(this.changeDetector);
+    }
+
+    addQuizBatch() {
+        if (!this.quizExercise.quizBatches) {
+            this.quizExercise.quizBatches = [];
+        }
+        this.quizExercise.quizBatches.push(new QuizBatch());
+    }
+
+    removeQuizBatch(quizBatch: QuizBatch) {
+        if (this.quizExercise.quizBatches) {
+            const idx = this.quizExercise.quizBatches.indexOf(quizBatch);
+            if (idx >= 0) {
+                this.quizExercise.quizBatches.splice(idx, 1);
+            }
+        }
+    }
+
+    /**
+     * Apply updates for changed course and quizExercise
+     * @param changes the changes to apply
+     */
+    ngOnChanges(changes: SimpleChanges): void {
+        if (changes.course || changes.quizExercise) {
+            this.init();
+        }
+    }
+
+    /**
+     * Update the categories and overwrite the cache, overwrites existing categories
+     * @param categories the new categories
+     */
+    updateCategories(categories: ExerciseCategory[]) {
+        this.quizExercise.categories = categories;
+        this.exerciseCategories = categories;
+        this.cacheValidation();
+    }
+
+    /**
+     * Determine which dropdown to display depending on the relationship between start time, end time, and current time
+     * @returns {string} Name of the dropdown to show
+     */
+    get showDropdown(): string {
+        if (!this.quizExercise || !this.quizExercise.quizStarted || this.isImport) {
+            return 'isVisibleBeforeStart';
+        } else if (this.quizExercise.quizEnded) {
+            return 'isOpenForPractice';
+        } else {
+            return 'active';
+        }
+    }
+
+    /**
+     * Returns whether pending changes are present, preventing a deactivation.
+     */
+    canDeactivate(): boolean {
+        return !this.pendingChangesCache;
+    }
+
+    /**
+     * displays the alert for confirming refreshing or closing the page if there are unsaved changes
```

Comment:
```suggestion
     * Displays the alert for confirming refreshing or closing the page if there are unsaved changes
```

Reply:
If that is fine for you, I will do this in a followup PR, I will have to do anyways.